geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/GeoJsonExporter.scala

```diff
@@ -8,19 +8,21 @@
 
 package org.locationtech.geomesa.tools.export.formats
 
-import java.io.{OutputStream, OutputStreamWriter}
+import java.io.OutputStreamWriter
 
+import com.google.gson.stream.JsonWriter
 import org.locationtech.geomesa.features.serialization.GeoJsonSerializer
-import org.locationtech.geomesa.tools.export.formats.FeatureExporter.{ByteCounter, ByteCounterExporter}
+import org.locationtech.geomesa.tools.`export`.formats.FeatureExporter.ExportStream
+import org.locationtech.geomesa.tools.export.formats.FeatureExporter.ByteCounterExporter
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
-class GeoJsonExporter(os: OutputStream, counter: ByteCounter) extends ByteCounterExporter(counter) {
-
-  private val writer = GeoJsonSerializer.writer(new OutputStreamWriter(os))
+class GeoJsonExporter(stream: ExportStream) extends ByteCounterExporter(stream) {
 
+  private var writer: JsonWriter = _
   private var serializer: GeoJsonSerializer = _
 
   override def start(sft: SimpleFeatureType): Unit = {
+    writer = GeoJsonSerializer.writer(new OutputStreamWriter(stream.os))
```

Comment:
Reading through this makes me wonder if we could use a lazy val for writer?

Also, this start method is not thread safe.  I am guessing we use it in a single threaded manner.

Reply:
yeah, all the exporters are single threaded. we could use a lazy val, but we'd still have to track whether the val was initialized. closing the writer will write the closing `}` to the json stream, so we don't want to instantiate it until we know that there is data to export, otherwise you'll end up with an empty feature collection but you'll still get an output file.